src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_IndexerAccess.cs

```diff
@@ -108,6 +108,8 @@ private BoundExpression VisitIndexerAccess(BoundIndexerAccess node, bool isLeftO
             BoundIndexerAccess? oldNodeOpt,
             bool isLeftOfAssignment)
         {
+            rewrittenReceiver = AdjustReceiverForExtensionsIfNeeded(rewrittenReceiver, indexer);
```

Comment:
We should probably have tests that `VisitArgumentsAndCaptureReceiverIfNeeded` behaves correctly with our rewritten receiver.

Reply:
There was a problem with interpolation handler arguments (fixed by adjusting the type of the interpolation "instance" placeholder)